lib/dry/cli/styles.rb

```diff
@@ -0,0 +1,187 @@
+      end
+
+      # Styled text
+      #
+      # @since 1.3.0
+      class StyledText
+        def initialize(text)
+          @text = text
+        end
+
+        # Makes `StyledText` printable
+        #
+        # @since 1.3.0
+        def to_s
+          @text
+        end
+
+        # since 1.3.0
+        def bold
+          chainable_update!("\e[#{BOLD}m#{@text}")
+        end
+
+        # since 1.3.0
+        def dim
+          chainable_update!("\e[#{DIM}m#{@text}")
+        end
+
+        # since 1.3.0
+        def italic
+          chainable_update!("\e[#{ITALIC}m#{@text}")
+        end
+
+        # since 1.3.0
+        def underline
+          chainable_update!("\e[#{UNDERLINE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def blink
+          chainable_update!("\e[#{BLINK}m#{@text}")
+        end
+
+        # since 1.3.0
+        def reverse
+          chainable_update!("\e[#{REVERSE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def invisible
+          chainable_update!("\e[#{INVISIBLE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def black
+          chainable_update!("\e[#{BLACK}m#{@text}")
+        end
+
+        # since 1.3.0
+        def red
+          chainable_update!("\e[#{RED}m#{@text}")
+        end
+
+        # since 1.3.0
+        def green
+          chainable_update!("\e[#{GREEN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def yellow
+          chainable_update!("\e[#{YELLOW}m#{@text}")
+        end
+
+        # since 1.3.0
+        def blue
+          chainable_update!("\e[#{BLUE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def magenta
+          chainable_update!("\e[#{MAGENTA}m#{@text}")
+        end
+
+        # since 1.3.0
+        def cyan
+          chainable_update!("\e[#{CYAN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def white
+          chainable_update!("\e[#{WHITE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_black
+          chainable_update!("\e[#{ON_BLACK}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_red
+          chainable_update!("\e[#{ON_RED}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_green
+          chainable_update!("\e[#{ON_GREEN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_yellow
+          chainable_update!("\e[#{ON_YELLOW}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_blue
+          chainable_update!("\e[#{ON_BLUE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_magenta
+          chainable_update!("\e[#{ON_MAGENTA}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_cyan
+          chainable_update!("\e[#{ON_CYAN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_white
+          chainable_update!("\e[#{ON_WHITE}m#{@text}")
+        end
+
+        private
+
+        # @since 1.3.0
+        # @api private
+        def chainable_update!(new_text)
```

Comment:
Instead of checking and only adding the `RESET` if one doesn't exist, could we just throw the `RESET` on at the end when it's coerced to a string in `to_s`? I think that should be fine. 

In that case, stylize(stylize(...)) might get two resets at the end, but that's a no-op anyway? We could also check in there and only add it if it doesn't already end with one.